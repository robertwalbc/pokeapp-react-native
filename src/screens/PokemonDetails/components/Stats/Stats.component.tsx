import React from 'react';
import { View, Text } from 'react-native';
import { styles } from './Stats.styles.ts';
import { StatBar } from '../StatBar';

interface Stat {
  base_stat: number;
  stat: {
    name: string;
  };
}

interface StatsProps {
  stats: Stat[];
}

export const Stats: React.FC<StatsProps> = ({ stats }) => (
  <>
    <Text style={styles.sectionTitle}>Stats</Text>
    <View style={styles.statsContainer}>
      {stats.map((stat, index) => (
        <View key={index} style={styles.statRow}>
          <Text style={styles.statName}>{stat.stat.name}</Text>
          <StatBar value={stat.base_stat} color="#4CAF50" />
          <Text style={styles.statValue}>{stat.base_stat}</Text>
        </View>
      ))}
    </View>
  </>
); 