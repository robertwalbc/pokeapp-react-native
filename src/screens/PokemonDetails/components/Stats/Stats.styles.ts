import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  sectionTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    marginTop: 16,
    marginBottom: 8,
  },
  statsContainer: {
    marginBottom: 16,
  },
  statRow: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 8,
  },
  statName: {
    width: 100,
    fontSize: 14,
  },
  statValue: {
    width: 40,
    textAlign: 'right',
    fontSize: 14,
  },
}); 