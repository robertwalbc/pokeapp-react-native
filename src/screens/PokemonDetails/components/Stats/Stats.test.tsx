import React from 'react';
import renderer from 'react-test-renderer';
import { Stats } from './Stats.component';

describe('Stats', () => {
  it('renders the section title and stat rows', () => {
    const stats = [
      { base_stat: 45, stat: { name: 'hp' } },
      { base_stat: 49, stat: { name: 'attack' } },
    ];
    const tree = renderer.create(<Stats stats={stats} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
}); 