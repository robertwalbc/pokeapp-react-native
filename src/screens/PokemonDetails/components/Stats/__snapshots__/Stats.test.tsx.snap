// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats renders the section title and stat rows 1`] = `null`;
